### PR TITLE
Make the log message more translation-friendly

### PR DESCRIPTION
**Pull request by cocoatomo**
Subject: Make the log message more translation-friendly

### Feature or Bugfix
- Bugfix

### Purpose
- Since current log message is not suitable for translation, so this PR organize log message templates in order to make the messages more translation-friendly.
- Since current log message may produce an incorrectly spelled word, so this PR fix that bug.

### Detail
- ~~Separate a gerund of a processing type from the log message~~
- ~~Make a gerund of a processing type translatable~~
- Separate a log message for read processing from for write processing

### Relates
- N/A

### CODE REVIEW

File: sphinx/application.py [hunk 1]
```diff
@@ -1133,18 +1133,20 @@ def is_parallel_allowed(self, typ):
         """
         if typ == 'read':
             attrname = 'parallel_read_safe'
+            gerund_typ = __("reading")
         elif typ == 'write':
             attrname = 'parallel_write_safe'
+            gerund_typ = __("writing")
```
Comment (tk0miya) [inline sphinx/application.py]: I feel `action` is better. What do you think?

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM!
Could you rebase this to 1.7-release branch please?

File: sphinx/application.py [hunk 2]
```diff
@@ -1133,18 +1133,20 @@ def is_parallel_allowed(self, typ):
...
 
         for ext in itervalues(self.extensions):
             allowed = getattr(ext, attrname, None)
             if allowed is None:
                 logger.warning(__("the %s extension does not declare if it is safe "
-                                  "for parallel %sing, assuming it isn't - please "
+                                  "for parallel %s, assuming it isn't - please "
                                   "ask the extension author to check and make it "
-                                  "explicit"), ext.name, typ)
+                                  "explicit"), ext.name, gerund_typ)
```
Comment (tk0miya) [inline sphinx/application.py]: Is it no problem to replace a single word in sentence for translation?
I guess the following style is better for translation. what do you think?
```
if typ == 'read':
    message = __("the %s extension..."
                 "for parallel reading, ...")
elif typ == 'write':
    message = __("the %s extension..."
                 "for parallel writing, ...")
else:
    ...
```

**Comment by cocoatomo** [comment]
(Edit)
- Rebase onto 1.7-release branch
- Avoid excessive code-sharing
- Update the comment

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Merged. Thank you for your contribution!

**Pull request merged**